### DISCUSSION TITLE
Fitur selectInput pada R Shiny

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#613] [Q&A] [answered]
Saya memiliki kodingan seperti ini : 

UI
<img width="483" alt="image" src="https://user-images.githubusercontent.com/106010284/229993518-372a0f4c-a779-441a-83b6-c6c377c1831b.png">

Server
<img width="544" alt="image" src="https://user-images.githubusercontent.com/106010284/229993611-a83dcf0f-a8b3-424c-96e8-26985afd0f8f.png">

Bagaimana caranya agar choice pada selectInput berisi unique dari data yang diupload, pada kolom "gabung" ?

Terima kasih sebelumnya 🙏

**Comment by Yusraf002** [comment]
Dear @yohanespm77 ,

Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Comment by rafiff23** [comment, marked as answer]
Coba ini mas, 
```
choices = unique(input_data$gabung)
```

**Reply by yohanespm77** [reply]
Gabisa mas, input_data not found. Di UI gada var input_data soalnya.

**Reply by rafiff23** [reply]
di bagian server tambahin ini mas
```
output$selectinput <- renderUI({
input_data <- readr::read_csv(input$filelSdatapath)
input_data <- janitor::clean_names(input_data)
input_data <- input_data %>%
filter (!status_code_custom %in% "0") %>%
filter Caction_custom %in% "alerted") %>%
mutate (policy_name_custom = gsub("/Common/",
mutate(start_time = mdy_hms(start_time)) %>%
mutate(day = day(start_time)) %>%
mutate (gabung = paste(attack_custom, policy_name_custom, sep = " -> "))

selectInput("selectinput", "Pilih Jenis Serangan", choices =unique(input_data$gabung))
})
```

di ui nya jadi begini
```
tabItem(
tabName = "jenis_serangan",
fluidrRow(
box (
width = 5,
uiOutput("selectinput")
 )
   )
     )
```

**Reply by yohanespm77** [reply]
Thanks mas @rafiff23
